lib/rules/no-undef.js

```diff
@@ -90,6 +90,9 @@ module.exports = function(context) {
             scope = scope.upper;
             variables = scope.variables.concat(variables);
         }
+        if (scope.childScopes.length) {
```

Comment:
You are right, but just above we loop across the scopes until reaching the global one, so at this point we already are in the global scope.